src/crush/CrushCompiler.cc

```diff
@@ -816,6 +816,49 @@ void CrushCompiler::dump(iter_t const& i, int ind)
 }
 
 
+//Sometimes when defined the bucket above the item which the bucket
```

Comment:
oic.

so the case is: 
```
rack using_foo { item foo }
host foo { ... }
```

if an item being used by a bucket is defined *after* that bucket. CRUSH compiler will create a map by which we can not identify that item when selecting in that bucket.